documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -293,6 +293,14 @@ JUnit Jupiter comes with many of the assertion methods that JUnit 4 has and adds
 that lend themselves well to being used with Java 8 lambdas. All JUnit Jupiter assertions
 are `static` methods in the `{Assertions}` class.
 
+Assertion methods can accept an optional third parameter, which can be either a `String` or a `Supplier<String>`.
+When using a `Supplier<String>` (e.g., a lambda expression) as the message supplier,
+the message is evaluated lazily.
+
+Lazily evaluating the assertion message can provide a performance benefit,
+especially if the message construction is complex or time-consuming,
+as it is only evaluated when the assertion fails.
```

Comment:
```suggestion
Assertion methods optionally accept the assertion message as their third parameter, which
can be either a `String` or a `Supplier<String>`.

When using a `Supplier<String>` (e.g., a lambda expression), the message is evaluated
lazily. This can provide a performance benefit, especially if message construction is
complex or time-consuming, as it is only evaluated when the assertion fails.
```

Reply:
You're welcome